coalib/results/SourceRange.py

```diff
@@ -10,7 +10,10 @@ class SourceRange(TextRange):
     @enforce_signature
     def __init__(self,
                  start: SourcePosition,
-                 end: (SourcePosition, None)=None):
```

Comment:
I think you're assuming a user will always use the from_* functions. That's only for easier usage.
A user can just create an instance of this class normally